lib/charms/layer/jenkins/packages.py

```diff
@@ -142,3 +145,22 @@ def jenkins_upgradable(self):
                 parse_version(self.jenkins_version())):
             return True
         return False
+
+    def clean_old_plugins(self):
+        """
+        Remove old plugins directories created by jenkins.deb and old versions
+        of this charm.
+        """
+        # Remove detached-plugins extracted from jenkins.deb
+        hookenv.log(
+            "Removing outdated detached plugins from jenkins.deb")
+        os.system("sudo rm -r /var/cache/jenkins/war/WEB-INF/detached-plugins")
+        # Remove old plugins from old charm versions
+        hookenv.log(
+            "Removing plugins from old charm versions")
+        for directory in glob("%s/*/" % paths.PLUGINS):
+            os.system("sudo rm -r %s" % directory)
+        # Remove plugin files with no version in its name
```

Comment:
Should this be a log entry instead of a comment (possibly for each item we're removing just before the os.system call)?